### PR TITLE
[ODBC] Support ODBC external table of SQLServer and revise the doc.

### LINKED ISSUE #6222: [New Feature] Support ODBC External Table Of Doris for SQLServer
```
CREATE EXTERNAL TABLE `baseall_sqlserver` (
  `k1` decimal(9, 3) NOT NULL COMMENT "",
  `k2` char(10) NOT NULL COMMENT "",
  `k3` datetime NOT NULL COMMENT "",
  `k5` varchar(20) NOT NULL COMMENT "",
  `k6` double NOT NULL COMMENT ""
) ENGINE=ODBC
COMMENT "ODBC"
PROPERTIES (
"host" = "192.168.0.1",
"port" = "8086",
"user" = "test",
"password" = "test",
"database" = "test",
"table" = "baseall",
"driver" = "SQL Server Driver",
"odbc_type" = "sqlserver"
);
```

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Support ODBC external table of SQLServer

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6222) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

File: docs/zh-CN/extending-doris/odbc-of-doris.md [hunk 1]
```diff
@@ -208,15 +208,32 @@ set enable_odbc_transcation = true;
 |   DATETIME  | DATETIME |  |
 |   NUMBER  | DECIMAL |  |
 
+### SQLServer
+
+|  MySQL  | Doris  |             替换方案              |
```
Comment (yangzhg) [inline docs/zh-CN/extending-doris/odbc-of-doris.md:213]: MYSQL ？

File: docs/en/extending-doris/odbc-of-doris.md [hunk 2]
```diff
@@ -212,15 +212,32 @@ There are different data types among different databases. Here, the types in eac
 |   DATETIME  | DATETIME |  |
 |   NUMBER  | DECIMAL |  |
 
+### SQLServer
+
+|  MySQL  | Doris  |             替换方案              |
```
Comment (yangzhg) [inline docs/en/extending-doris/odbc-of-doris.md:217]: Mysql ？

File: docs/en/extending-doris/odbc-of-doris.md [hunk 3]
```diff
@@ -212,15 +212,32 @@ There are different data types among different databases. Here, the types in eac
...
+| :------: | :----: | :-------------------------------: |
+|  BOOLEAN  | BOOLEAN  |                         |
+|   CHAR   |  CHAR  |            Only UTF8 encoding is supported            |
+| VARCHAR | VARCHAR |       Only UTF8 encoding is supported      |
```
Comment (yangzhg) [inline docs/en/extending-doris/odbc-of-doris.md:221]: no varchar in sqlserver

File: docs/en/extending-doris/odbc-of-doris.md [hunk 4]
```diff
@@ -214,7 +214,7 @@ There are different data types among different databases. Here, the types in eac
 
 ### SQLServer
 
-|  MySQL  | Doris  |             替换方案              |
+|  SQLServer  | Doris  |             替换方案              |
```
Comment (yangzhg) [inline docs/en/extending-doris/odbc-of-doris.md:217]: Chinese characters in en docs

Comment (yangzhg) [reply docs/en/extending-doris/odbc-of-doris.md:217]:
```suggestion
|  SQLServer  | Doris  |             Alternation rules              |
```

**Review by morningman — APPROVED** [review]

**Pull request merged**